bundles/core/org.eclipse.smarthome.core.thing/src/main/java/org/eclipse/smarthome/core/thing/internal/ThingManager.java

```diff
@@ -236,6 +242,11 @@ public Thing run() {
             }
         }
 
+        @Override
+        public void changeThingType(Thing thing, ThingTypeUID thingTypeUID, Configuration configuration) {
+            migrateThingType(thing, thingTypeUID, configuration);
```

Comment:
Why introducing an extra method here that is only called from this place? You could put the code right in here!